src/WebJobs.Script/Rpc/FunctionRegistration/FunctionDispatcher.cs

```diff
@@ -282,7 +266,7 @@ public async void WorkerRestart(WorkerRestartEvent workerRestart)
 
         private async Task DisposeAndRestartWorkerChannel(string runtime, string workerId)
         {
-            bool isWebHostChannel = await _webHostLanguageWorkerChannelManager.ShutdownChannelIfExistsAsync(runtime, workerId);
```

Comment:
This is problematic and can hang, waiting for this "await"

Reply:
PR: https://github.com/Azure/azure-functions-host/pull/4877 handles setting exception on all the tasks that started